realm/realm-library/src/main/cpp/io_realm_internal_SharedRealm.cpp

```diff
@@ -0,0 +1,314 @@
+#include <object-store/src/object_store.hpp>
+#include "io_realm_internal_SharedRealm.h"
+
+#include "shared_realm.hpp"
+#include "util.hpp"
+
+using namespace realm;
+
+JNIEXPORT jlong JNICALL
+Java_io_realm_internal_SharedRealm_nativeCreateConfig
+(JNIEnv *env, jclass, jstring realm_path, jbyteArray key, jboolean read_only, jboolean in_memory,
+ jboolean cache, jboolean disable_format_upgrade, jboolean auto_change_notification) {
+    TR_ENTER()
+
+    JStringAccessor path(env, realm_path);
+    Realm::Config *config = new Realm::Config();
+    config->path = path;
+    config->encryption_key = jbytearray_to_vector(env, key);
+    config->read_only = read_only;
+    config->in_memory = in_memory;
+    config->cache = cache;
+    config->disable_format_upgrade = disable_format_upgrade;
+    config->automatic_change_notifications = auto_change_notification;
+    return reinterpret_cast<jlong>(config);
+}
+
+JNIEXPORT void JNICALL
+Java_io_realm_internal_SharedRealm_nativeCloseConfig
+(JNIEnv *, jclass, jlong config_ptr) {
+    TR_ENTER_PTR(config_ptr)
+    delete RC(config_ptr);
+}
+
+JNIEXPORT jlong JNICALL
+Java_io_realm_internal_SharedRealm_nativeGetSharedRealm
+(JNIEnv *env, jclass, jlong config_ptr) {
+    TR_ENTER_PTR(config_ptr)
+
+    try {
+        SharedRealm shared_realm = Realm::get_shared_realm(*RC(config_ptr));
+        return reinterpret_cast<jlong>(new SharedRealm(shared_realm));
```

Comment:
Using `std::move(shared_realm)` here would avoid the refcount churn that the copy currently causes.